### PR TITLE
Bump minimum verions

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref https://github.com/pandas-dev/pandas/pull/26595#issuecomment-500934897

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -100,6 +100,28 @@ Other Enhancements
 - :class:`DatetimeIndex` and :class:`TimedeltaIndex` now have a ``mean`` method (:issue:`24757`)
 - :meth:`DataFrame.describe` now formats integer percentiles without decimal point (:issue:`26660`)
 
+.. _whatsnew_0250.dependencies:
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: can you merge this with the existing section
```
Increased minimum versions for dependencies
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v0.25.0.rst]: these as part of the numpy 1.13 bump

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.0.rst]: Whoops forgot about that.

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -415,11 +417,13 @@ Optional libraries below the lowest tested version may still work, but are not c
 +-----------------+-----------------+
 | Package         | Minimum Version |
 +=================+=================+
+| BeautifulSoup4  | 4.6.0           |
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: isn't this beautifulsoup4 ? (lowercase)

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -401,6 +401,8 @@ If installed, we now require:
 +-----------------+-----------------+----------+
 | pytz            | 2015.4          |    X     |
 +-----------------+-----------------+----------+
+| python-dateutil | 2.6.1           +   X
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: can you space over one (i know this is pedantic, but otherwise won't be see right away)

**Comment by TomAugspurger** [comment]
All green.

File: doc/source/install.rst [hunk 4]
```diff
@@ -220,13 +220,19 @@ installed), make sure you have `pytest
...
 ------------
 
-* `setuptools <https://setuptools.readthedocs.io/en/latest/>`__: 24.2.0 or higher
-* `NumPy <http://www.numpy.org>`__: 1.13.3 or higher
-* `python-dateutil <https://dateutil.readthedocs.io/en/stable/>`__: 2.5.0 or higher
-* `pytz <http://pytz.sourceforge.net/>`__: 2015.4 or higher
+================================================================ ==========================
+Package                                                          Minimum supported version
+================================================================ ==========================
+`setuptools <https://setuptools.readthedocs.io/en/latest/>`__    24.2.0
```
Comment (jreback) [inline doc/source/install.rst]: we should bump this too, but orthogonal

**Review by jreback — APPROVED** [review]

**Pull request merged**